neuralop/layers/embeddings.py

```diff
@@ -137,35 +162,117 @@ def forward(self, data, batched=True):
+    ----------
+    in_channels : int
+        Number of input channels to embed
+    num_freqs : int, optional
+        Number of frequencies in positional embedding.
+        By default, set to the number of input channels
+    embedding : {'nerf', 'transformer'}
+        Type of embedding to apply. For a function with N input channels, 
+        each channel value p is embedded via a function g with 2L channels 
+        such that g(p) is a 2L-dim vector. For 0 <= k < L:
+        
+        * 'nerf' : NERF-style encoding.  
+
+            g(p)_k = sin(2^(k) * Pi * p) if i is even
+
+                   = cos(2^(k) * Pi * p) if i is odd
+
+        * 'transformer' for transformer-style encoding.
+            Let z = `p / max_positions`. Then:
+
+            g(p)_k = sin((p / max_positions) ^ {2k / N}) if i is even
+
+                   = cos((p / max_positions) ^ {2k / N}) if i is odd
+    max_positions : int, optional
+        Maximum number of positions for the encoding, default 10000
+        Only used if `embedding == transformer`.
+
+    References
+    -----------
+    .. _[1]: 
+
+    Vaswani, A. et al (2017)
+        "Attention Is All You Need". 
+        NeurIPS 2017, https://proceedings.neurips.cc/paper_files/paper/2017/file/3f5ee243547dee91fbd053c1c4a845aa-Paper.pdf. 
+
+    .. _[2]: 
+    
+    Mildenhall, B. et al (2020)
+        "NeRF: Representing Scenes as Neural Radiance Fields for View Synthesis".
+        ArXiv, https://arxiv.org/pdf/2003.08934. 
+    """
+    def __init__(self, 
+                 in_channels: int,
+                 num_frequencies: int=None, 
+                 embedding_type: str='nerf', 
+                 max_positions: int=10000):
         super().__init__()
-        self.num_channels = num_channels
+        self.in_channels = in_channels
+        self.num_frequencies = num_frequencies
+        
+        # verify embedding type
+        allowed_embeddings = ['nerf', 'transformer']
+        assert embedding_type in allowed_embeddings, \
+            f"Error: embedding_type expected one of {allowed_embeddings}, received {embedding_type}"
+        self.embedding_type = embedding_type
         self.max_positions = max_positions
-        self.endpoint = endpoint
+    
+    
+    @property
+    def out_channels(self):
+        """
+        out_channels: required property for linking/composing model layers 
+        """
+        return 2 * self.num_frequencies * self.in_channels
 
     def forward(self, x):
-        freqs = torch.arange(
-            start=0, end=self.num_channels // 2, dtype=torch.float32, device=x.device
-        )
-        freqs = freqs / (self.num_channels // 2 - (1 if self.endpoint else 0))
-        freqs = (1 / self.max_positions) ** freqs
-        x = x.ger(freqs.to(x.dtype))
-        x = torch.cat([x.cos(), x.sin()], dim=1)
-        return x
-
+        """
+        Parameters 
+        -----------
+        x: torch.Tensor, shape (n_in, out_channels) or (batch, n_in, out_channels)
```

Comment:
Just small issues in annotation, it should be in_channels here instead of out_channels. It might be clearer to also clarify the return here.